src/core/search/searchdatatransformer.js

```diff
@@ -31,16 +32,44 @@ export default class SearchDataTransformer {
   }
 
   static transformVertical (data, formatters, verticalKey) {
-    // TODO (tmeyer): SPR-1923, centralize the allResultsForVertical parsing here.
+    const resultsContext = SearchDataTransformer._determineResultsContext(data.response);
+    const response = SearchDataTransformer._formVerticalResponseWithContext(data.response, resultsContext);
     return {
-      [StorageKeys.QUERY_ID]: data.response.queryId,
+      [StorageKeys.QUERY_ID]: response.queryId,
       [StorageKeys.NAVIGATION]: new Navigation(), // Vertical doesn't respond with ordering, so use empty nav.
-      [StorageKeys.VERTICAL_RESULTS]: VerticalResults.from(data.response, formatters, verticalKey),
-      [StorageKeys.DYNAMIC_FILTERS]: DynamicFilters.from(data.response),
-      [StorageKeys.INTENTS]: SearchIntents.from(data.response.searchIntents),
-      [StorageKeys.SPELL_CHECK]: SpellCheck.from(data.response.spellCheck),
-      [StorageKeys.ALTERNATIVE_VERTICALS]: AlternativeVerticals.from(data.response, formatters),
-      [StorageKeys.LOCATION_BIAS]: LocationBias.from(data.response.locationBias)
+      [StorageKeys.VERTICAL_RESULTS]: VerticalResults.from(response, formatters, verticalKey, resultsContext),
+      [StorageKeys.DYNAMIC_FILTERS]: DynamicFilters.from(response.facets),
+      [StorageKeys.INTENTS]: SearchIntents.from(response.searchIntents),
+      [StorageKeys.SPELL_CHECK]: SpellCheck.from(response.spellCheck),
+      [StorageKeys.ALTERNATIVE_VERTICALS]: AlternativeVerticals.from(response, formatters),
+      [StorageKeys.LOCATION_BIAS]: LocationBias.from(response.locationBias)
     };
   }
+
+  /**
+   * Determine the {@link ResultsContext} of the given vertical results.
+   * @param {Object} response
+   */
+  static _determineResultsContext (response) {
+    const hasResults = response.results && response.resultsCount > 0;
+    return hasResults ? ResultsContext.NORMAL : ResultsContext.NO_RESULTS;
+  }
+
+  /**
+   * Form response as if the results from `allResultsForVertical` were the actual
+   * results in `results`
+   * @param {Object} response The server response
+   */
+  static _formVerticalResponseWithContext (response, resultsContext) {
```

Comment:
I think this name is kind of confusing. Maybe parseVerticalResponse? I think it could also simplify things a bit if we made resultsContext part of our VerticalResults model. So, this method would return an instance of VerticalResults that included a resultsContext property. Since resultsContext doesn't make sense without a VerticalResults and we need a resultsContext to interpret VerticalResults, I think it makes sense to add resultsContext to the VerticalResults model.

Reply:
The other models don't currently take in the VerticalResults model, they take in things like response.facets and response.searchIntents. resultsContext is also already part of the VerticalResults model, are you saying we want something like a VerticalResponse model? So basically move this _parseVerticalResponse logic into a class?

Reply:
Gotcha. Yeah, I think we should have the VerticalResponse (created here) have resultsContext. I think we should also rename this method to make its purpose clearer.

Reply:
ah okay that makes a lot more sense than making it a special property, even though it's the only one that property that doesn't come directly from the response, updated!